From 02a68c009433840913b46791555190c25c9d97e8 Mon Sep 17 00:00:00 2001
From: Christophe Parant <c.parant@phytec.fr>
Date: Mon, 27 Apr 2026 14:46:28 +0200
Subject: [PATCH] configs: phytec stm32mp: update env offset for firmware
 update

The current default u-boot environment offsets doesn't match the
default partitionning used for TF-A firmware update feature.
Configure the offsets for this feature.

Signed-off-by: Christophe Parant <c.parant@phytec.fr>
---
 configs/phycore-stm32mp13_defconfig | 4 ++--
 configs/phycore-stm32mp15_defconfig | 4 ++--
 configs/phyflex_stm32mp23_defconfig | 4 ++--
 configs/phyflex_stm32mp25_defconfig | 4 ++--
 4 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/configs/phycore-stm32mp13_defconfig b/configs/phycore-stm32mp13_defconfig
index 569c80e1742..34583d1f3e8 100644
--- a/configs/phycore-stm32mp13_defconfig
+++ b/configs/phycore-stm32mp13_defconfig
@@ -4,7 +4,7 @@ CONFIG_ARCH_STM32MP=y
 CONFIG_TFABOOT=y
 CONFIG_SYS_MALLOC_F_LEN=0x180000
 CONFIG_CUSTOM_SYS_INIT_SP_ADDR=0xc0400000
-CONFIG_ENV_OFFSET=0x480000
+CONFIG_ENV_OFFSET=0x900000
 CONFIG_ENV_SECT_SIZE=0x40000
 CONFIG_DEFAULT_DEVICE_TREE="stm32mp135f-phycore-som-1gib"
 CONFIG_DEFAULT_FDT_FILE="oftree"
@@ -12,7 +12,7 @@ CONFIG_STM32MP13X=y
 CONFIG_DDR_CACHEABLE_SIZE=0x8000000
 CONFIG_CMD_STM32KEY=y
 CONFIG_TARGET_PHYTEC_PHYCORE_STM32MP13X=y
-CONFIG_ENV_OFFSET_REDUND=0x4C0000
+CONFIG_ENV_OFFSET_REDUND=0x940000
 CONFIG_CMD_STM32PROG=y
 # CONFIG_ARMV7_NONSEC is not set
 CONFIG_SYS_LOAD_ADDR=0xc2000000
diff --git a/configs/phycore-stm32mp15_defconfig b/configs/phycore-stm32mp15_defconfig
index e2c360432ec..bec818ba619 100644
--- a/configs/phycore-stm32mp15_defconfig
+++ b/configs/phycore-stm32mp15_defconfig
@@ -3,14 +3,14 @@ CONFIG_ARCH_STM32MP=y
 CONFIG_TFABOOT=y
 CONFIG_SYS_MALLOC_F_LEN=0x80000
 CONFIG_CUSTOM_SYS_INIT_SP_ADDR=0xc0100000
-CONFIG_ENV_OFFSET=0x480000
+CONFIG_ENV_OFFSET=0x900000
 CONFIG_ENV_SECT_SIZE=0x40000
 CONFIG_DEFAULT_DEVICE_TREE="stm32mp157c-phycore-som-1gib"
 CONFIG_DEFAULT_FDT_FILE="oftree"
 CONFIG_DDR_CACHEABLE_SIZE=0x8000000
 CONFIG_CMD_STM32KEY=y
 CONFIG_TARGET_PHYTEC_PHYCORE_STM32MP15X=y
-CONFIG_ENV_OFFSET_REDUND=0x4C0000
+CONFIG_ENV_OFFSET_REDUND=0x940000
 CONFIG_CMD_STM32PROG=y
 # CONFIG_ARMV7_NONSEC is not set
 CONFIG_SYS_LOAD_ADDR=0xc2000000
diff --git a/configs/phyflex_stm32mp23_defconfig b/configs/phyflex_stm32mp23_defconfig
index 9b183d0161a..68ced5eac99 100644
--- a/configs/phyflex_stm32mp23_defconfig
+++ b/configs/phyflex_stm32mp23_defconfig
@@ -3,13 +3,13 @@ CONFIG_USE_ARCH_MEMCPY=y
 CONFIG_ARCH_STM32MP=y
 CONFIG_SYS_MALLOC_F_LEN=0x500000
 CONFIG_CUSTOM_SYS_INIT_SP_ADDR=0x90000000
-CONFIG_ENV_OFFSET=0x480000
+CONFIG_ENV_OFFSET=0x900000
 CONFIG_ENV_SECT_SIZE=0x40000
 CONFIG_DEFAULT_DEVICE_TREE="stm32mp235c-phyflex-fpsc-g-som"
 CONFIG_DEFAULT_FDT_FILE="oftree"
 CONFIG_STM32MP23X=y
 CONFIG_CMD_STM32KEY=y
-CONFIG_ENV_OFFSET_REDUND=0x4C0000
+CONFIG_ENV_OFFSET_REDUND=0x940000
 CONFIG_TARGET_PHYTEC_PHYFLEX_STM32MP23X=y
 CONFIG_CMD_STM32PROG=y
 CONFIG_SYS_LOAD_ADDR=0x84000000
diff --git a/configs/phyflex_stm32mp25_defconfig b/configs/phyflex_stm32mp25_defconfig
index 366815548f0..13e5fd8b2df 100644
--- a/configs/phyflex_stm32mp25_defconfig
+++ b/configs/phyflex_stm32mp25_defconfig
@@ -3,13 +3,13 @@ CONFIG_USE_ARCH_MEMCPY=y
 CONFIG_ARCH_STM32MP=y
 CONFIG_SYS_MALLOC_F_LEN=0x500000
 CONFIG_CUSTOM_SYS_INIT_SP_ADDR=0x90000000
-CONFIG_ENV_OFFSET=0x480000
+CONFIG_ENV_OFFSET=0x900000
 CONFIG_ENV_SECT_SIZE=0x40000
 CONFIG_DEFAULT_DEVICE_TREE="stm32mp257f-phyflex-fpsc-g-som"
 CONFIG_DEFAULT_FDT_FILE="oftree"
 CONFIG_STM32MP25X=y
 CONFIG_CMD_STM32KEY=y
-CONFIG_ENV_OFFSET_REDUND=0x4C0000
+CONFIG_ENV_OFFSET_REDUND=0x940000
 CONFIG_TARGET_PHYTEC_PHYFLEX_STM32MP25X=y
 CONFIG_CMD_STM32PROG=y
 CONFIG_SYS_LOAD_ADDR=0x84000000
-- 
2.34.1

